### PR TITLE
lang: use phantomdata in account type instead of '_account: T'

### LINKED ISSUE #1140: lang: adjust program account to use `PhantomData` to indicate that its generic is not used in its field
currently it has a `_account: T` field to do this. `PhantomData` is the idiomatic way of showing that `T` is not required inside the fields of the struct

### PR DESCRIPTION
**Pull request by paul-schaaf**
fixes #1140

### CODE REVIEW

File: lang/src/accounts/program.rs [hunk 1]
```diff
@@ -75,13 +71,7 @@ impl<'a, T: Id + AccountDeserialize + Clone> Program<'a, T> {
             None
         };
 
-        // Programs have no data so use an empty slice.
-        let mut empty = &[][..];
-        Ok(Program::new(
-            info.clone(),
-            T::try_deserialize(&mut empty)?,
```
Comment (armaniferrante) [inline lang/src/accounts/program.rs:82]: We should remove this trait bound and the associated implementations as well.

Comment (paul-schaaf) [reply lang/src/accounts/program.rs:82]: the `Deserialize` trait bound for `T`?

Comment (paul-schaaf) [reply lang/src/accounts/program.rs:82]: yea makes sense

Comment (paul-schaaf) [reply lang/src/accounts/program.rs:82]: the `dex` struct also implements `AccountSerialize`. Was that intentional?

Comment (armaniferrante) [reply lang/src/accounts/program.rs:82]: Sounds like a mistake. Can remove that too.

Comment (paul-schaaf) [reply lang/src/accounts/program.rs:82]: done

**Pull request merged**